readthedocs/projects/views/private.py

```diff
@@ -407,44 +408,50 @@ def get_queryset(self):
-    """Project subprojects view and form view"""
-    project = get_object_or_404(Project.objects.for_admin_user(request.user),
-                                slug=project_slug)
+class ProjectRelationshipMixin(ProjectAdminMixin, PrivateViewMixin):
 
-    form_kwargs = {
-        'parent': project,
-        'user': request.user,
-    }
-    if request.method == 'POST':
-        form = SubprojectForm(request.POST, **form_kwargs)
-        if form.is_valid():
-            form.save()
-            broadcast(type='app', task=tasks.symlink_subproject, args=[project.pk])
-            project_dashboard = reverse(
-                'projects_subprojects', args=[project.slug])
-            return HttpResponseRedirect(project_dashboard)
-    else:
-        form = SubprojectForm(**form_kwargs)
+    model = ProjectRelationship
+    form_class = ProjectRelationshipForm
+    lookup_field = 'child__slug'
+    lookup_url_kwarg = 'subproject_slug'
 
-    subprojects = project.subprojects.all()
+    def get_queryset(self):
+        self.project = self.get_project()
+        return self.model.objects.filter(parent=self.project)
 
-    return render_to_response(
-        'projects/project_subprojects.html',
-        {'form': form, 'project': project, 'subprojects': subprojects},
-        context_instance=RequestContext(request)
-    )
+    def get_form(self, data=None, files=None, **kwargs):
+        kwargs['user'] = self.request.user
+        return super(ProjectRelationshipMixin, self).get_form(data, files, **kwargs)
 
+    def form_valid(self, form):
+        broadcast(type='app', task=tasks.symlink_subproject,
+                  args=[self.get_project().pk])
+        return super(ProjectRelationshipMixin, self).form_valid(form)
 
-@login_required
-def project_subprojects_delete(request, project_slug, child_slug):
-    parent = get_object_or_404(Project.objects.for_admin_user(request.user), slug=project_slug)
-    child = get_object_or_404(Project.objects.all(), slug=child_slug)
-    parent.remove_subproject(child)
-    broadcast(type='app', task=tasks.symlink_subproject, args=[parent.pk])
-    return HttpResponseRedirect(reverse('projects_subprojects',
-                                        args=[parent.slug]))
+    def get_success_url(self):
+        return reverse('projects_subprojects', args=[self.get_project().slug])
+
+
+class ProjectRelationshipList(ProjectRelationshipMixin, ListView):
+
+    def get_context_data(self, **kwargs):
+        ctx = super(ProjectRelationshipList, self).get_context_data(**kwargs)
+        ctx['superproject'] = self.project.superprojects.first()
+        return ctx
+
+
+class ProjectRelationshipCreate(ProjectRelationshipMixin, CreateView):
+    pass
+
+
+class ProjectRelationshipUpdate(ProjectRelationshipMixin, UpdateView):
+    pass
+
+
+class ProjectRelationshipDelete(ProjectRelationshipMixin, DeleteView):
+
+    def get(self, request, *args, **kwargs):
+        return self.http_method_not_allowed(request, *args, **kwargs)
```

Comment:
Nope, get on deleteview displays a form that asks for confirmation. I'd like to settle on a deleteview override and never use confirmation forms, but we do use them in some places currently.